great_expectations/analytics/events.py

```diff
@@ -255,6 +256,31 @@ def __init__(self, validation_definition_id: str | None = None):
         )
 
 
+@dataclass
+class ValidationDefinitionRanEvent(_ValidationDefinitionEvent):
+    checkpoint_id: str | None = None
+    _allowed_actions: ClassVar[List[Action]] = [VALIDATION_DEFINITION_RAN]
+
+    @override
+    def _properties(self) -> dict:
+        return {
+            "checkpoint_id": self.checkpoint_id,
+            "validation_definition_id": self.validation_definition_id,
+        }
+
+    def __init__(
+        self,
+        *,
+        validation_definition_id: str | None = None,
+        checkpoint_id: str | None = None,
+    ):
+        self.checkpoint_id = checkpoint_id
+        super().__init__(
```

Comment:
Coming back to this, it's a bummer that we are implementing our own constructors for these events. dataclasses should do that for us. Seems like there's a lot of low-priority cleanup around here that we could get to once we finish implementing all the features 😛 .